libpod/container.go

```diff
@@ -1072,7 +1073,14 @@ func (c *Container) CGroupPath() (string, error) {
 	case define.SystemdCgroupsManager:
 		if rootless.IsRootless() {
 			uid := rootless.GetRootlessUID()
-			return filepath.Join(c.config.CgroupParent, fmt.Sprintf("user-%d.slice/user@%d.service/user.slice", uid, uid), createUnitName("libpod", c.ID())), nil
+			parts := strings.SplitN(c.config.CgroupParent, "/", 2)
+
+			dir := ""
+			if len(parts) > 1 {
+				dir = parts[1]
+			}
+
+			return filepath.Join(parts[0], fmt.Sprintf("user-%d.slice/user@%d.service/user.slice/%s", uid, uid, dir), createUnitName("libpod", c.ID())), nil
 		}
 		return filepath.Join(c.config.CgroupParent, createUnitName("libpod", c.ID())), nil
```

Comment:
It seems like we should be setting c.config.CgroupParent to the value you're computing when we make the container? That would fix this and probably make other things more accurate.

Reply:
isn't the `CgroupParent` option controllable by the user?

The issue is just with the special naming convention we (and OCI runtimes as well) use for systemd

Reply:
I think that if the user decides to override the default with something that doesn't work, it's not our fault. Might want to add a warning to the manpages that rootless requires a systemd slice delegated to your user?

Reply:
the value generated by default for pods is something like:
`user.slice/user-libpod_pod_180b780d58fcde353f8bac1d22f2d401ece580df743ea3f36717366bb2a332ce.slice`

Should it be handled or do we change the default?

Reply:
As I look at this more... this looks like it would need createUnitName changes, which makes it much less attractive.

I'd like to start making rootless-specific support more generic and less of a special case, but this looks like a bad place to start.